traefik.go

```diff
@@ -1,16 +1,130 @@
 package main
 
 import (
+	"encoding/json"
+	log "github.com/Sirupsen/logrus"
+	"github.com/containous/flaeg"
+	"github.com/containous/staert"
+	"github.com/containous/traefik/middlewares"
+	"github.com/containous/traefik/provider"
 	fmtlog "log"
+	"net/http"
 	"os"
+	"reflect"
 	"runtime"
+	"strings"
 )
 
 func main() {
 	runtime.GOMAXPROCS(runtime.NumCPU())
-	if err := traefikCmd.Execute(); err != nil {
+
+	//traefik config inits
+	traefikConfiguration := NewTraefikConfiguration()
+	traefikPointersConfiguration := NewTraefikPointersConfiguration()
+	//traefik Command init
+	traefikCmd := &flaeg.Command{
+		Name: "traefik",
+		Description: `traefik is a modern HTTP reverse proxy and load balancer made to deploy microservices with ease.
+Complete documentation is available at https://traefik.io`,
+		Config:                traefikConfiguration,
+		DefaultPointersConfig: traefikPointersConfiguration,
+		Run: func() error {
+			run(traefikConfiguration)
+			return nil
+		},
+	}
+
+	//version Command init
+	versionCmd := &flaeg.Command{
+		Name:                  "version",
+		Description:           `Print version`,
+		Config:                struct{}{},
+		DefaultPointersConfig: struct{}{},
+		Run: func() error {
+			fmtlog.Println(Version + " built on the " + BuildDate)
+			return nil
+		},
+	}
+
+	//init flaeg source
+	f := flaeg.New(traefikCmd, os.Args[1:])
+	//add custom parsers
+	f.AddParser(reflect.TypeOf(EntryPoints{}), &EntryPoints{})
+	f.AddParser(reflect.TypeOf(DefaultEntryPoints{}), &DefaultEntryPoints{})
+	f.AddParser(reflect.TypeOf(provider.Namespaces{}), &provider.Namespaces{})
+	//add version command
+	f.AddCommand(versionCmd)
+	if _, err := f.Parse(traefikCmd); err != nil {
 		fmtlog.Println(err)
 		os.Exit(-1)
 	}
+
+	//staert init
+	s := staert.NewStaert(traefikCmd)
+	//init toml source
+	toml := staert.NewTomlSource("traefik", []string{traefikConfiguration.ConfigFile, "/etc/traefik/", "$HOME/.traefik/", "."})
+
+	//add sources to staert
+	s.AddSource(toml)
+	s.AddSource(f)
+	if _, err := s.GetConfig(); err != nil {
```

Comment:
I guess this populates something in the staert `s` but the name feels weird, `Get…` something makes me think I'm gonna get something in return, and there we don't care. I'm think it's just about validate the config, but I'm not sure :sweat_smile: